includes/promoted-jobs/class-wp-job-manager-promoted-jobs-notifications.php

```diff
@@ -0,0 +1,335 @@
+		];
+	}
+
+	/**
+	 * Extract required job listing meta from the $_POST array.
+	 *
+	 * @param WP_Post $post Post object.
+	 */
+	private function get_post_data( $post ) {
+		$post_data = [];
+		foreach ( $this->watched_fields as $field ) {
+			if ( 'job_description' === $field ) {
+				$post_data[ $field ] = $post->post_content;
+				continue;
+			}
+			if ( 'job_title' === $field ) {
+				$post_data[ $field ] = $post->post_title;
+				continue;
+			}
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Missing -- Input sanitized in registered post meta config;
+			$post_data[ $field ] = isset( $_POST[ "_$field" ] ) ? wp_unslash( $_POST[ "_$field" ] ) : '';
+		}
+		return $post_data;
+	}
+
+	/**
+	 * Determine if any job field has changed.
+	 *
+	 * @param array $current_values Current values of the job.
+	 * @param array $new_values New values of the job.
+	 */
+	private function post_has_changed( $current_values, $new_values ) {
+		foreach ( $current_values as $key => $value ) {
+			if ( $value !== $new_values[ $key ] ) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Check if a job is promoted.
+	 *
+	 * @param int $post_id Post ID.
+	 * @return bool
+	 */
+	private function is_promoted_job( $post_id ) {
+		return get_post_meta( $post_id, '_promoted', true );
+	}
+
+	/**
+	 * Check if we should notify wpjobmanager.com of a change.
+	 *
+	 * @param int $post_id Post ID.
+	 * @return bool
+	 */
+	public function should_notify( $post_id ) {
+		if ( 'job_listing' !== get_post_type( $post_id ) ) {
+			return false;
+		}
+		if ( ! $this->is_promoted_job( $post_id ) ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Notify wpjobmanager.com when a Job Listing is trashed.
+	 *
+	 * @param int $post_id Post ID.
+	 * @return void
+	 */
+	public function promoted_job_trashed( $post_id ) {
+		if ( ! $this->should_notify( $post_id ) ) {
+			return;
+		}
+		$this->send_notification();
+	}
+
+	/**
+	 * Notify wpjobmanager.com when a Job Listing is updated.
+	 *
+	 * @param int   $post_id Post ID.
+	 * @param array $values Values.
+	 * @return void
+	 */
+	public function promoted_job_updated( $post_id, $values ) {
+		if ( ! $this->should_notify( $post_id ) ) {
+			return;
+		}
+
+		if ( $this->post_has_changed( $this->get_watched_fields_meta( $post_id ), $this->map_to_watched_fields( $values ) ) ) {
+			$this->send_notification();
+		}
+	}
+
+	/**
+	 * Notify wpjobmanager.com when a Job Listing is updated on admin.
+	 * Values are available in the $_POST global, sanitized and validated.
+	 *
+	 * @param int     $post_id Post ID.
+	 * @param WP_Post $post Post object.
+	 * @return void
+	 */
+	public function promoted_job_updated_admin( $post_id, $post ) {
+		if ( ! $this->should_notify( $post_id ) ) {
+			return;
+		}
+
+		if ( $this->post_has_changed( $this->get_watched_fields_meta( $post_id ), $this->get_post_data( $post ) ) ) {
```

Comment:
For some reason, it's always returning with no changes for me. In the `post_has_changed` it receives the new values in both arguments.

Let me know if you can't reproduce the issue here, so I can help you to debug. 😉

Reply:
Never mind about this, should work now in 7ebc3e1f265fa7c6b7f4995c47483d896fcdb845